mitodl/hugo-course-publisher
# Resource Search

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #208 

#### What's this PR do?
- Adds a UI option to search for resources instead of courses
- Changes the query generation code to handle resource search
- Changes `SearchResult` component to handle both courses and resources

#### How should this be manually tested?
- Check out https://github.com/mitodl/open-discussions/pull/3159 if it hasn't been merged yet and (re-)index some OCW courses
- Go to the search page.  Course search should be enabled by default and should work as before.
- Click 'Resources' just under the search term bar.  The results should update to show resource results instead of course results.  There should be a mix of pdf's, web pages, and videos, but nothing else.
- The course title under each result should link to the course page (probably a 404 locally, try the URL path on ocw-beta to make sure it works)
- The resource title for a Youtube video should link to the video; for a PDF should link to a relative URL that should work on ocw-beta via CDN, for pages should link to a relative URL that works on ocw-beta.


#### Any background context you want to provide?
There is a hardcoded filter on `content_type` to pdf, page, or video.  Eventually this will probably be a full-blown facet.

Per the issue, the following are not yet included:
- thumbnails for web pages, non-youtube videos
- facets

#### Screenshots (if appropriate)

RESOURCES:


<img width="950" alt="Screen Shot 2020-09-24 at 4 22 44 PM" src="https://user-images.githubusercontent.com/187676/94274802-2269a880-ff14-11ea-92cb-a203b2f8109d.png">


COURSES:


<img width="954" alt="Screen Shot 2020-09-24 at 4 22 34 PM" src="https://user-images.githubusercontent.com/187676/94274821-2ac1e380-ff14-11ea-8e2d-f6a31151b521.png">
